tracer/src/Datadog.Trace/ClrProfiler/AutomaticTracer.cs

```diff
@@ -23,19 +23,23 @@ SpanContext IDistributedTracer.GetSpanContext()
             {
                 return null;
             }
-            else if (DistributedTrace.Value is SpanContext spanContext)
+
+            var value = DistributedTrace.Value;
+
+            if (value is SpanContext spanContext)
             {
                 return spanContext;
             }
-            else
-            {
-                return SpanContextPropagator.Instance.Extract(DistributedTrace.Value);
-            }
+
+            return SpanContextPropagator.Instance.Extract(value);
         }
 
         void IDistributedTracer.SetSpanContext(SpanContext value)
         {
-            DistributedTrace.Value = value;
+            if (_child != null)
```

Comment:
Might be worth cross referencing your explanation below?
```suggestion
            // This is a performance optimisation. See comment in GetDistributedTrace() about potential race condition
            if (_child != null)
```